glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -105,42 +103,7 @@ open class GleanInternalAPI internal constructor () {
     internal var isSendingToTestEndpoint: Boolean = false
 
     // Store the build information provided by the application.
-    internal var buildInfo: BuildInfo? = null
-
-    /**
-     * Initialize the Glean SDK.
-     *
-     * This should only be initialized once by the application, and not by
-     * libraries using the Glean SDK. A message is logged to error and no
-     * changes are made to the state if initialize is called a more than
-     * once.
-     *
-     * A LifecycleObserver will be added to send pings when the application goes
-     * into foreground and background.
-     *
-     * This method must be called from the main thread.
-     *
-     * This form of `initialize` is deprecated. Use the form that requires a
-     * `buildInfo` parameter instead.
-     *
-     * @param applicationContext [Context] to access application features, such
-     * as shared preferences
-     * @param uploadEnabled A [Boolean] that determines whether telemetry is enabled.
-     *     If disabled, all persisted metrics, events and queued pings (except
-     *     first_run_date and first_run_hour) are cleared.
-     * @param configuration A Glean [Configuration] object with global settings.
-     */
-    @JvmOverloads
-    @Synchronized
-    @MainThread
-    @Deprecated("The buildInfo parameter will be required in the future. See bug 1691953")
-    fun initialize(
```

Comment:
Not sure we need to worry right away, but did you test how this affects (if at all) Java consumers?

Reply:
Java consumers can't leave out the config parameter anymore: https://github.com/mozilla/glean/pull/1752/files#diff-55350221eb40c8d32d4c30f8392220d105c0a44fa9ccbbd338b975f09acdb6cdR42